src/commandline_options.cpp

```diff
@@ -558,28 +556,33 @@ commandline_options::commandline_options(const std::vector<std::string>& args)
 
 void commandline_options::parse_log_domains_(const std::string &domains_string, const int severity)
 {
-	const std::vector<std::string> domains = utils::split(domains_string, ',');
-	for (const std::string& domain : domains)
+	if(std::vector<std::string> domains = utils::split(domains_string, ','); !domains.empty())
 	{
-		if (!log)
-			log = std::vector<std::pair<int, std::string>>();
-		log->emplace_back(severity, domain);
+		if(!log)
+		{
+			log.emplace();
+		}
+		for (auto&& domain : domains)
+		{
+			assert(log && "`log` must have been initialized before this loop");
```

Comment:
I think that assert can never trigger.

In general, things should handle the situation rather than assert.

Reply:
Asserts are useful to verify some preconditions

Maybe it's tio much here, but I wanted to still offer the check .at was offering but without a cost at -O3

Reply:
just realized I replied thinking this comment was on something else

I'll remove the assert